Remove deprecated InstructTemplate from llm_pte_finetuning example

RdoubleA opened the pull request:
See #6552 for context.

Here, we remove the InstructTemplate classes and instead directly replace with a dataset builder that uses the necessary torchtune data components. The associated config is also updated.

Closes #6552.

larryliu0820 reviewed (approved):
Stamping since this is blocking. Please fix internal job failure

RdoubleA committed:
Remove deprecated InstructTemplate from llm_pte_finetuning example (#6557)

Summary:
See https://github.com/pytorch/executorch/issues/6552 for context.

Here, we remove the InstructTemplate classes and instead directly replace with a dataset builder that uses the necessary torchtune data components. The associated config is also updated.


Reviewed By: larryliu0820

Differential Revision: D65168404

Pulled By: RdoubleA
